### PR TITLE
Add keyless to rust sdk and txn-emitter

### PR DESCRIPTION
**Pull request by heliuchuan**
## Description

This implements a bare sdk for Keyless (no pepper/proof fetching) and adds the ability to use keyless accounts for the txn-emitter.

The emitter requires a proof file which contains serialized ZeroKnowledgeSignatures.

The following args are added to specify the proper parameters for accounts generation

```
    #[clap(long, value_parser = ConfigKey::<Ed25519PrivateKey>::from_encoded_string, requires = "proof_file_path", requires = "keyless_jwt")]
    pub keyless_ephem_secret_key: Option<ConfigKey<Ed25519PrivateKey>>,

    #[clap(long)]
    pub proof_file_path: Option<String>,

    #[clap(long, default_value_t = 2700000000)]
    pub epk_expiry_date_secs: u64,

    #[clap(long)]
    pub keyless_jwt: Option<String>,
```

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [x] Tests

## Which Components or Systems Does This Change Impact?
- [x] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [x] I tested both happy and unhappy path of the functionality
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: crates/transaction-emitter-lib/src/emitter/local_account_generator.rs [hunk 1]
```diff
@@ -64,3 +69,66 @@ impl LocalAccountGenerator for PrivateKeyAccountGenerator {
         Ok(accounts)
     }
 }
+
+pub struct KeylessAccountGenerator {
+    serialized_proofs: Vec<String>,
+    ephemeral_secret_key: Ed25519PrivateKey,
+    expiry_date_secs: u64,
+}
+
```
Comment (sitalkedia) [inline crates/transaction-emitter-lib/src/emitter/local_account_generator.rs:78]: Let's add support for reading a proof file - the file path can be specified as env variable and initialize the `serialized_proofs` ?

File: sdk/src/types.rs [hunk 2]
```diff
@@ -516,50 +516,75 @@ pub struct KeylessAccount {
     #[allow(dead_code)]
     pepper: Pepper,
     zk_sig: ZeroKnowledgeSig,
-    jwt: String,
+    jwt_header_json: String,
+    #[allow(dead_code)]
+    jwt: Option<String>,
```
Comment (zjma) [inline sdk/src/types.rs:521]: so why not delete?

File: sdk/src/types.rs [hunk 3]
```diff
@@ -516,50 +516,75 @@ pub struct KeylessAccount {
...
     }
 
-    pub fn get_jwt_header_json(&self) -> String {
-        let parts: Vec<&str> = self.jwt.split('.').collect();
+    pub fn derive_account(
+        jwt: &str,
+        ephemeral_key_pair: EphemeralKeyPair,
+        pepper: Option<Pepper>,
+        zk_sig: Option<ZeroKnowledgeSig>,
+    ) -> Result<Self> {
```
Comment (zjma) [inline sdk/src/types.rs:558]: this function returns a `Result` (so letting the caller handle errors i assume) but also has `unwrap`s and `[idx]` which can panic directly, by design?

File: sdk/src/types.rs [hunk 4]
```diff
@@ -516,50 +516,75 @@ pub struct KeylessAccount {
...
+        let parts: Vec<&str> = jwt.split('.').collect();
         let header_bytes = base64::decode(parts[0]).unwrap();
-        String::from_utf8(header_bytes).unwrap()
+        let jwt_header_json = String::from_utf8(header_bytes).unwrap();
+        let jwt_payload_json = base64::decode_config(parts[1], base64::URL_SAFE).unwrap();
+        let claims: Claims = serde_json::from_slice(&jwt_payload_json)?;
+
+        let uid_key = "sub".to_owned();
```
Comment (zjma) [inline sdk/src/types.rs:565]: i forgot how `uid_key` is determined: isn't it possible for it to be `email` as well?

File: crates/transaction-emitter-lib/src/emitter/test_proofs_for_localnet_txn_emitter.txt [hunk 5]
```diff
@@ -0,0 +1 @@
+todo: populate this
```
Comment (zjma) [inline crates/transaction-emitter-lib/src/emitter/test_proofs_for_localnet_txn_emitter.txt:1]: so if this file has n lines, then we can create no more than n accounts for txn emitter,  which is a reason we want proof simulator?

File: sdk/src/types.rs (hunk 4 above)
Comment (heliuchuan) [reply sdk/src/types.rs:565]: it is.  added an arg to allow for specifying

File: sdk/src/types.rs (hunk 2 above)
Comment (heliuchuan) [reply sdk/src/types.rs:521]: fixed - used

File: sdk/src/types.rs (hunk 3 above)
Comment (heliuchuan) [reply sdk/src/types.rs:558]: not by design, fixed

File: crates/transaction-emitter-lib/src/emitter/test_proofs_for_localnet_txn_emitter.txt (hunk 5 above)
Comment (heliuchuan) [reply crates/transaction-emitter-lib/src/emitter/test_proofs_for_localnet_txn_emitter.txt:1]: for local, it helps.  for devnet and testnet idk

File: sdk/src/types.rs (hunk 3 above)
Comment (zjma) [reply sdk/src/types.rs:558]: L560 `parts[0]` can still panic 😆

Comment (heliuchuan) [reply sdk/src/types.rs:558]: got me

**Review by zjma — APPROVED** [review]
pre-approved, assuming test_proofs_for_localnet_txn_emitter.txt will be populated to pass the updated tests.

File: crates/transaction-emitter-lib/src/args.rs [hunk 6]
```diff
@@ -218,6 +218,18 @@ pub struct EmitArgs {
...
+    pub keyless_ephem_secret_key: Option<ConfigKey<Ed25519PrivateKey>>,
+
+    #[clap(long)]
+    pub proof_file_path: Option<String>,
+
+    #[clap(long, default_value_t = 2700000000)]
+    pub epk_expiry_date_secs: u64,
+
+    #[clap(long)]
+    pub keyless_jwt: Option<String>,
```
Comment (igor-aptos) [inline crates/transaction-emitter-lib/src/args.rs:232]: let's put all these params inside AccountType::Keyless, so they cannot be set if keyless is not used, and need to be set if keyless is used.

File: crates/transaction-emitter-lib/src/args.rs [hunk 7]
```diff
@@ -234,6 +246,12 @@ pub struct CreateAccountsArgs {
     /// used and printed.
     #[clap(long)]
     pub account_minter_seed: Option<String>,
+
+    #[clap(long)]
+    pub keyless_jwt: Option<String>,
+
+    #[clap(long)]
+    pub proof_file_path: Option<String>,
```
Comment (igor-aptos) [inline crates/transaction-emitter-lib/src/args.rs:254]: let's add `pub account_type: AccountType` here instead?

File: crates/transaction-emitter-lib/src/emitter/mod.rs [hunk 8]
```diff
@@ -188,6 +191,15 @@ pub struct EmitJobRequest {
...
     account_minter_seed: Option<[u8; 32]>,
+
+    // Arguments for Keyless Load Testing
+    keyless_ephem_secret_key: Option<[u8; 32]>,
+
+    proof_file_path: Option<String>,
+
+    epk_expiry_date_secs: Option<u64>,
+
+    keyless_jwt: Option<String>,
```
Comment (igor-aptos) [inline crates/transaction-emitter-lib/src/emitter/mod.rs:202]: you can remove this if it's part of AccountType

**Review by igor-aptos — CHANGES_REQUESTED** [review]
thanks looks great! minor config cleanup, and this is ready!

File: crates/transaction-emitter-lib/src/emitter/mod.rs (hunk 8 above)
Comment (heliuchuan) [reply crates/transaction-emitter-lib/src/emitter/mod.rs:203]: couldn't restructure into the enum

File: crates/transaction-emitter-lib/src/args.rs (hunk 6 above)
Comment (heliuchuan) [reply crates/transaction-emitter-lib/src/args.rs]: talked offline, grouped the args into struct as there's no way to do this for enum without subcommand it seems...

**Review by igor-aptos — APPROVED** [review]

**Pull request merged**